digiwf-engine/digiwf-engine-cockpit/src/main/java/de/muenchen/oss/digiwf/cockpit/security/camunda/OAuthIdentityServiceProvider.java

```diff
@@ -0,0 +1,307 @@
+package de.muenchen.oss.digiwf.cockpit.security.camunda;
+
+
+import de.muenchen.oss.digiwf.spring.security.SecurityConfiguration;
+import org.apache.commons.lang3.StringUtils;
+import org.camunda.bpm.engine.identity.*;
+import org.camunda.bpm.engine.impl.GroupQueryImpl;
+import org.camunda.bpm.engine.impl.Page;
+import org.camunda.bpm.engine.impl.TenantQueryImpl;
+import org.camunda.bpm.engine.impl.UserQueryImpl;
+import org.camunda.bpm.engine.impl.identity.ReadOnlyIdentityProvider;
+import org.camunda.bpm.engine.impl.interceptor.CommandContext;
+import org.camunda.bpm.engine.impl.persistence.AbstractManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import de.muenchen.oss.digiwf.spring.security.JwtClaims;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * This Identity Provider returns only the currently logged-in user's data. It is required for the camunda apps to work properly with OAuth
+ * login. Only the necessary methods of the interface are implemented, most methods just return null or empty things.
+ **/
+public class OAuthIdentityServiceProvider extends AbstractManager implements ReadOnlyIdentityProvider {
+
+  @Override
+  public User findUserById(String userId) {
+    return createUserQuery().userId(userId).singleResult();
+  }
+
+  @Override
+  public UserQuery createUserQuery() {
+    return new OAuthUserQueryImpl(this);
+  }
+
+  @Override
+  public UserQuery createUserQuery(CommandContext commandContext) {
+    return new OAuthUserQueryImpl(this);
+  }
+
+  @Override
+  public NativeUserQuery createNativeUserQuery() {
+    return null;
+  }
+
+  @Override
+  public boolean checkPassword(String userId, String password) {
+    return false;
+  }
+
+  @Override
+  public Group findGroupById(String groupId) {
+    return createGroupQuery().groupId(groupId).singleResult();
+  }
+
+  @Override
+  public GroupQuery createGroupQuery() {
+    return new OAuthGroupQueryImpl(this);
```

Comment:
Do we "only" use token roles for authorization? Do we need ldap here or a step before?

Reply:
yes. why should we use LDAP groups for that? OUs are representing the org structure and should not be used for giving you permission to application IMHO. So, let's stick on SSO roles.